tensorflow/cc/gradients/nn_grad_test.cc

```diff
@@ -113,5 +113,23 @@ TEST_F(NNGradTest, SeluGrad) {
   RunTest(x, x_init_value, y, shape);
 }
 
+TEST_F(NNGradTest, BiasAddGradHelper) {
+  TensorShape shape({4, 5});
+  TensorShape bias_shape({5});
+  auto x1 = Placeholder(scope_, DT_FLOAT, Placeholder::Shape(shape));
+  Tensor bias1  = test::AsTensor<float>(
+    {-0.9f, -0.5f, 0.1f, 0.3f, 0.5f},
+    {5});
+  auto y1 = BiasAdd(scope_, x1, bias1);
+  auto x2  = test::AsTensor<float>(
+    {-0.9f, -0.7f, -0.5f, -0.3f, -0.1f, 0.1f, 0.3f, 0.5f, 0.7f, 0.9f,
+     -0.9f, -0.7f, -0.5f, -0.3f, -0.1f, 0.1f, 0.3f, 0.5f, 0.7f, 0.9f},
+    {4, 5});
+  auto bias2 = Placeholder(scope_, DT_FLOAT, Placeholder::Shape(bias_shape));
+  auto y2 = BiasAdd(scope_, x2, bias2);
+  RunTest(x1, shape, y1, shape);
```

Comment:
You can create another overload for RunTest that takes multiple "x_shapes" like here in array_grad_test: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/cc/gradients/array_grad_test.cc#L44

This should allow you to just specify shapes for the inputs and run one RunTest. Here is an example of a test that does that: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/cc/gradients/array_grad_test.cc#L56